qa/rpc-tests/test_framework/util.py

```diff
@@ -750,7 +750,10 @@ def disconnect_all(node):
     """Disconnect all peers from the passed node"""
     peers = node.getpeerinfo()
     for p in peers:
-        node.disconnectnode(p["addr"])
+        try:
+            node.disconnectnode(p["addr"])
+        except JSONRPCException as e:  # exception thrown if node is already disconnected so ignore since the intention is to disconnect
+            pass
```

Comment:
If there's an exception thrown , why not examine the text for the message you would expect if there was no node connected which would be "Node not found in connected nodes".

Reply:
Great find by the way!

Reply:
you are right that that would be more accurate.  However, it would also be sensitive to changes to the string which is intended for humans so may require more maintenance.  This is why I didn't add that check but I'm ok either way.  Let me know.

Reply:
good point, let's leave it as is.

Reply:
What about adding a logging debug statement that report the error string?